clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -300,6 +299,7 @@ public abstract class StoreIngestionTask implements Runnable, Closeable {
   protected final ChunkAssembler chunkAssembler;
   private final Optional<ObjectCacheBackend> cacheBackend;
   private final DaVinciRecordTransformer recordTransformer;
+  private InternalDaVinciRecordTransformer internalRecordTransformer;
```

Comment:
Do we even need to downcast like this?  Also, do we need both a reference to recordTransformer and internalRecordTransformer?

Reply:
I'm unclear what downcast means here, but you're right we only need one reference to recordTransformer.

Reply:
downcast in code means that we're using a stricter derived class vs. a base class (DavinciRecordTransformer vs. InternalDavinciRecordTransformer).  It's not technically wrong to do it like this and there are scenarios where it makes sense to do so, but theres nothing about the interface of InternalDVRT that we rely on, and it can make things a bit more inflexible when it comes to mocking/testing/changing the code.